src/frontend/src/views/Data/index.tsx

```diff
@@ -37,25 +37,26 @@ const Data: FunctionComponent = () => {
 
   const router = useRouter();
 
+  const sampleResponse = useSampleInfo();
   const treeResponse = useTreeInfo();
-  const { data, isLoading } = treeResponse;
+  const { data: sampleData, isLoading: isSampleInfoLoading } = sampleResponse;
+  const { data: treeData, isLoading: isTreeInfoLoading } = treeResponse;
 
   useEffect(() => {
     const setBioinformaticsData = async () => {
       setIsDataLoading(true);
-      if (isLoading) return;
-      const sampleResponse = await fetchSamples();
+      if (isTreeInfoLoading || isSampleInfoLoading) return;
       setIsDataLoading(false);
 
-      const apiSamples = sampleResponse["samples"];
-      const apiTrees = data?.phylo_trees;
-
+      const apiSamples = sampleData?.samples ?? [];
       setSamples(apiSamples);
```

Comment:
Do we really need this state stuff for `setSamples` or `setTrees` anymore? It seems unnecessary, to have state keeping a copy of what react-query fetched, feels like we could probably just use react-query `apiSamples` and `apiTrees` directly below rather than intermediating it with React component state. (Of course, this is me effectively asking you to do _more_ refactoring, so I'm totally fine with you not putting even more on your plate.)